sklearn/cluster/optics_.py

```diff
@@ -0,0 +1,545 @@
+# -*- coding: utf-8 -*-
+"""Ordering Points To Identify the Clustering Structure (OPTICS)
+
+These routines execute the OPTICS algorithm, and implement various
+cluster extraction methods of the ordered list.
+
+Authors: Shane Grigsby <refuge@rocktalus.com>
+         Amy X. Zhang <axz@mit.edu>
+License: BSD 3 clause
+"""
+
+import scipy as sp
+import numpy as np
+from ..utils import check_array
+from ..neighbors import BallTree
+from ..base import BaseEstimator, ClusterMixin
+from ..metrics.pairwise import pairwise_distances
+from ._optics_inner import quick_scan
+
+
+class SetOfObjects(BallTree):
```

Comment:
if we decide to modify this one later, we don't have to deprecate if it is private